test/client/audio_consumer_test.py

```diff
@@ -1,171 +1,144 @@
-
-        TOLERANCE_RANGE_FRAMES = (
-            WakewordExtractor.MAX_ERROR_SECONDS * audio.sample_rate *
-            audio.sample_width)
+        tolerance = 4000
+        ideal_begin = 70000
+        ideal_end = 92000
 
         monitor = {}
-        self.recognizer.set_transcriptions(
-            ["what's the weather next week", ""])
+        self.recognizer.set_transcriptions(["what's the weather next week"])
 
         def wakeword_callback(message):
             monitor['pos_begin'] = message.get('pos_begin')
             monitor['pos_end'] = message.get('pos_end')
 
         self.loop.once('recognizer_loop:wakeword', wakeword_callback)
-        self.consumer.try_consume_audio()
+        self.consumer.read_audio()
 
-        pos_begin = monitor.get('pos_begin')
-        self.assertIsNotNone(pos_begin)
-        diff = abs(pos_begin - TRUE_POS_BEGIN)
+        actual_begin = monitor.get('pos_begin')
+        self.assertIsNotNone(actual_begin)
+        diff = abs(actual_begin - ideal_begin)
         self.assertTrue(
-            diff <= TOLERANCE_RANGE_FRAMES,
-            str(diff) + " is not less than " + str(TOLERANCE_RANGE_FRAMES))
+                diff <= tolerance,
+                str(diff) + " is not less than " + str(tolerance))
 
-        pos_end = monitor.get('pos_end')
-        self.assertIsNotNone(pos_end)
-        diff = abs(pos_end - TRUE_POS_END)
+        actual_end = monitor.get('pos_end')
+        self.assertIsNotNone(actual_end)
+        diff = abs(actual_end - ideal_end)
         self.assertTrue(
-            diff <= TOLERANCE_RANGE_FRAMES,
-            str(diff) + " is not less than " + str(TOLERANCE_RANGE_FRAMES))
+                diff <= tolerance,
+                str(diff) + " is not less than " + str(tolerance))
 
     def test_wakeword_in_beginning(self):
-        self.queue.put(self.__create_sample_from_test_file('mycroft'))
+        self.queue.put(self.__create_sample_from_test_file('weather_mycroft'))
+        self.recognizer.set_transcriptions(["what's the weather next week"])
         monitor = {}
-        self.recognizer.set_transcriptions([
-            "what's the weather next week", ""])
 
         def callback(message):
             monitor['utterances'] = message.get('utterances')
 
         self.loop.once('recognizer_loop:utterance', callback)
-        self.consumer.try_consume_audio()
+        self.consumer.read_audio()
+
         utterances = monitor.get('utterances')
         self.assertIsNotNone(utterances)
         self.assertTrue(len(utterances) == 1)
         self.assertEquals("what's the weather next week", utterances[0])
 
-    def test_wakeword_in_phrase(self):
+    def test_wakeword(self):
         self.queue.put(self.__create_sample_from_test_file('mycroft'))
+        self.recognizer.set_transcriptions(["silence"])
         monitor = {}
-        self.recognizer.set_transcriptions([
-            "he can do other stuff too", "what's the weather in cincinnati"])
```

Comment:
There is no real audio testing this 2 phrases and `mycroft` wakeword alone should not result in any remote STT query, because it'll only produce silence after the `word extraction` process. In practice, this test had exactly the same behaviour of the test: `test_wakeword_in_beginning` which uses a suitable audio test with a sentence including the wakeword.